### PR TITLE
Sort courses into Templates, Current, and Past on course mgmt page

### PR DESCRIPTION
**Pull request by sadleb**
### Description
The list of courses and templates on the Course Management page was getting unwieldy. This PR automatically
sorts courses that are not in a Current or Future Salesforce Program into a "Past Launched Courses" section at the bottom. Those that are in a Current or Future Program are now in their own section just below the templates.

Task: https://app.asana.com/0/1174274412967132/1200663885163758

### Screenshots (if appropriate):
Page now has three sections: Templates, Current, and Past.
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: spec/factories/salesforce_current_and_future_programs.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@
     sequence(:Id) { program_id }
     sequence(:Name) { |i| "TEST: Program#{i}" }
     sequence(:Canvas_Cloud_Accelerator_Course_ID__c) { canvas_course_id }
+    sequence(:Canvas_Cloud_LC_Playbook_Course_ID__c) { canvas_course_id + 1000 }
```
Comment (sadleb) [inline spec/factories/salesforce_current_and_future_programs.rb:31]: This is hacky, but it was a pain to change this and have all callers pass an array of `[accelerator course id, lc playbook id]` pairs.

File: app/views/courses/_list.html.erb [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
-    </ol>
-    <% if button_label %>
-      <div class="col d-flex justify-content-end">
-        <%= link_to "Add #{name}", new_course_path, class: "btn btn-lg btn-secondary m-0", role: "button" %>
-      </div>
-    <% end %>
-  </nav>
+  <div class="col">
+    <nav aria-label="breadcrumb" class="breadcrumb col mt-2">
+      <p class="breadcrumb-item active"><%= name %></p>
```
Comment (sadleb) [inline app/views/courses/_list.html.erb:4]: Just removing the `<ol>` -> not sure why it was in there, but all three sections had a prefix of 1. E.g. `1. Course Templates`, `1. Current Launched Courses`, etc.

File: app/services/grade_modules.rb [hunk 3]
```diff
@@ -18,13 +18,12 @@ def run
...
-      if programs['records'].empty?
-        Rails.logger.info("Exit early: no current/future accelerator programs")
+      canvas_course_ids = SalesforceAPI.client.get_current_and_future_accelerator_canvas_course_ids
+      span.add_field('app.grade_modules.canvas_course_ids', canvas_course_ids)
+      if canvas_course_ids.empty?
+        Rails.logger.info("Exit early: no current/future accelerator programs with a Canvas course ID set.")
         return
       end
-      canvas_course_ids = programs['records'].map { |r| r['Canvas_Cloud_Accelerator_Course_ID__c'].to_s }
-      span.add_field('app.grade_modules.canvas_course_ids', canvas_course_ids)
```
Comment (sadleb) [inline app/services/grade_modules.rb:27]: Have a look at the changes in the `lib/salesforce_api` first. I cleaned it up a little to not require consumers to parse out the `programs['records']` key and added two utility methods to just get the list of relevant Canvas course IDs. Most of the changes in this PR are just that refactoring.

File: app/views/courses/index.html.erb [hunk 4]
```diff
@@ -4,15 +4,22 @@
...
-  courses: @courses.launched_courses,
+  name: 'Current Launched Courses',
+  courses: @current_launched_courses,
+  button_label: nil,
+) %>
+
+<%= render(
+  'list',
+  name: 'Past Launched Courses',
+  courses: @past_launched_courses,
```
Comment (sadleb) [inline app/views/courses/index.html.erb:22]: This and the setting of these instance variables in the controller is the meat of this PR.

**Review by rshipp — APPROVED** [review]

**Pull request merged**